From 0eb6b8d4fc06c75bf44612c32d8fbe2b75a28180 Mon Sep 17 00:00:00 2001
From: Alexandru Ardelean <alexandru.ardelean@analog.com>
Date: Tue, 4 Sep 2018 16:05:55 +0300
Subject: [PATCH 16/25] dma: axi-dmac: don't check the number of frames for
 alignment

It doesn't look like the 2D transfers were used much, prior to commit
81b810593ebcf1a537042704c8363bbaafbcba0b, so this may have not received
much attention.

There is no requirement for Y_LENGTH to be aligned to the bus-width (or
anything). X_LENGTH is required to be aligned though.

So, we shouldn't check that the number of frames is aligned.

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>
(cherry picked from commit 04231a9a3ff27630f3453b5feca197723a5f3e88)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/dma/dma-axi-dmac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/dma-axi-dmac.c b/drivers/dma/dma-axi-dmac.c
index a6de75b6d954..8bbfd10a7274 100644
--- a/drivers/dma/dma-axi-dmac.c
+++ b/drivers/dma/dma-axi-dmac.c
@@ -577,7 +577,7 @@ static struct dma_async_tx_descriptor *axi_dmac_prep_interleaved(
 
 	if (chan->hw_2d) {
 		if (!axi_dmac_check_len(chan, xt->sgl[0].size) ||
-		    !axi_dmac_check_len(chan, xt->numf))
+		    xt->numf == 0)
 			return NULL;
 		if (xt->sgl[0].size + dst_icg > chan->max_length ||
 		    xt->sgl[0].size + src_icg > chan->max_length)
-- 
2.17.0

